### PR TITLE
DTM0: added configuration parameters for DTM0 service

### PR DESCRIPTION
**Pull request by just-now**

### CODE REVIEW

File: dtm0/Kbuild.sub [hunk 1]
```diff
@@ -3,4 +3,6 @@ m0tr_objects += \
                   dtm0/clk_src.o \
                   dtm0/clk_src_xc.o \
                   dtm0/tx_desc.o \
-                  dtm0/tx_desc_xc.o
+                  dtm0/tx_desc_xc.o \
+		  dtm0/service.o \
+		  dtm0/fop.o
```
Comment (ivan-alekhin) [inline dtm0/Kbuild.sub:8]: we need to check the "tabs vs. spaces" issue here. If there are tabs then let's replace them with spaces to be aligned with the rest of the codebase.

File: fop/fom_generic.c [hunk 2]
```diff
@@ -96,11 +97,9 @@ int32_t m0_rpc_item_generic_reply_rc(const struct m0_rpc_item *reply)
 }
 M0_EXPORTED(m0_rpc_item_generic_reply_rc);
 
-#include "reqh/reqh_service.h"
 static bool fom_is_update(const struct m0_fom *fom)
 {
-	/* XXX: make dtm0 service happy on client side */
-	return fom->fo_service->rs_service_fid.f_key != 0x1a &&
+	return !m0_dtm0_is_a_volatile_dtm(fom->fo_service) &&
```
Comment (ivan-alekhin) [inline fop/fom_generic.c:102]: s/!m0_dtm0_is_a_volatile_dtm/m0_dtm0_is_a_persistent_dtm/

File: dtm0/service.c [hunk 3]
```diff
@@ -223,16 +268,35 @@ static void dtm0_service_fini(struct m0_reqh_service *service)
...
+
+M0_INTERNAL bool m0_dtm0_is_a_volatile_dtm(struct m0_reqh_service *service)
+{
+	return service->rs_type->rst_typecode == M0_CST_DTM0 &&
+		origin(service) == DTM0_ON_VOLATILE;
+}
+
+M0_INTERNAL bool m0_dtm0_is_a_persistent_dtm(struct m0_reqh_service *service)
+{
+		return service->rs_type->rst_typecode == M0_CST_DTM0 &&
```
Comment (ivan-alekhin) [inline dtm0/service.c:296]: Indentation level seems to be wrong here. One '\t' is enough.

**Review by ivan-alekhin — APPROVED** [review]
the patch looks good. I added a few nitpicks.

File: fop/fom_generic.c (hunk 2 above)
Comment (just-now) [reply fop/fom_generic.c:102]: This part is correct due to 
`m0_dtm0_is_a_persistent_dtm() != !m0_dtm0_is_a_volatile_dtm()`

File: dtm0/service.c (hunk 3 above)
Comment (just-now) [reply dtm0/service.c:296]: fixed.

File: dtm0/Kbuild.sub (hunk 1 above)
Comment (just-now) [reply dtm0/Kbuild.sub:8]: fixed

**Pull request merged**